Update boot.asm

franzageek opened the pull request:
- Refactor the code (split it into .text and .data section);
- Improve label naming;
- Remove jumps before 'ret's.

<!--
List all changes made in this PR. Example:

* Fixed typos in README.md
* Updated badges
* Fixed hyperlinks
-->

<!-- Reference issues. https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue -->

<!-- Please check for duplicates before contributing -->

adamperkowski reviewed (changes requested):
Hey. Thanks for contributing again!
The code looks good but there's one small issue. It doesn't boot. I'll look into it later today.

franzageek commented:
Oop so sorry, lemme try to fix it. It could be that no entry point was specified, I'll give it a try.

adamperkowski commented:
I'll add a boot check to this [assembly](https://github.com/adamperkowski/highlightos/actions/workflows/asm.yml) CI, because now it only compiles and that could be misleading.

franzageek commented:
Sounds like a good idea. I'm on a Mac rn so I couldn't test the code before committing, thinking that because the changes were not invasive, it should've booted just fine. I've just updated it - lmk if it works fine :)

franzageek commented:
I think I've found the bug, trying to fix it.

adamperkowski commented:
Added a job that checks if a DOS boot sector is present. It's not a full "boot check" but it'll do.
https://github.com/adamperkowski/highlightos/blob/4346236180e39512335f32ba9e25e9a4fff452e7/.github/workflows/asm.yml#L26-L30

adamperkowski commented:
I'll update your branch.

franzageek commented:
There you go, fixed. I forgot that 16 bit Real Mode assembly does not support splitting code in text/data sections. I'm used to program in 32/64 bit Protected mode. Very sorry, should be fixed now.

While debugging, I found another bug - after pressing enter the program flow remained trapped in an infinite loop, because I replaced a jump with a ret.

 Fixed that as well.

adamperkowski commented:
Seems like you forgot about this https://github.com/adamperkowski/highlightos/pull/18/commits/bc369ce4a1a3c4ba9efd12798dd64bc55f794e3e .
Added.

franzageek commented:
Actually no - the reason was the `jmp shell` line on line 116/117.

adamperkowski commented:
Thanks for taking your time and contributing again. Much appreciated. I'll merge this if you don't want to add/change anything.

adamperkowski commented:
No problem at all. Happens. So yeah thanks for taking your time and contributing again. Much appreciated. I'll merge this if you don't want to add anything.

franzageek commented:
Sure! Really like the vision & the direction of this project.
Let me review the code - will add anything if necessary.

adamperkowski commented:
Message me privately please through Discord, Signal, e-mail or whatever you like. Merging.

franzageek commented:
Alright - just sent you a DM request on Discord.
